src/extensions/content_downloaders/curl_downloader/CMakeLists.txt

```diff
@@ -1,23 +1,22 @@
-project (curl_content_downloader)
-
+set (target_name curl_content_downloader)
```

Comment:
For all extension, I think we should keep them 'project' instead. Since we may (eventually) split the package. And we want to manage the .SO file version individually.

Reply:
We can still split it into a *.so. That's a target property not a project property. A project is the sum of all of the buildable targets within it. 

If we want to break it out into a shared library we'd be changing this to either a separate folder/repo from this project if we were to build it. 

The only time you traditionally have nested projects is when the higher level projects have no need of the targets built under neath it. 

That's the difference between a project vs target. 

If we wanted to change this to a shared library we'd need to change the add_library type from STATIC to DYNAMIC and CMake will handle the rest. It'll even handle the linking between the libraries as we go.